### ISSUE TITLE
copy-related-tag can not work on snapshot if any EBS volume does not exist

### ISSUE THREAD

**Issue by zbw1983** [cloud-custodian/cloud-custodian#5375]
I try to use copy-related-tag to add TAG for snapshot, below is my policy: 

policies:
    - name: copy-tags-from-ebs-volume-to-snapshot
      resource: ebs-snapshot
      actions:
        - type: copy-related-tag
          resource: ebs
          skip_missing: True
          key: VolumeId
          tags:
            - project

I have hundreds of snapshots, some volumes of them are already deleted, then when I run the policy, it reports below error. 

```
error:An error occurred (InvalidVolume.NotFound) when calling the DescribeVolumes operation: The volume 'vol-e8bd2ae8' does not exist.
2020-02-24 13:18:01,623 - custodian.actions - INFO - Tagged 0 resources from related, missing-skipped 878 unchanged 0
2020-02-24 13:18:01,623 - custodian.policy - INFO - policy:copy-tags-from-ebs-volume-to-snapshot action:copyrelatedresourcetag resources:878 execution_time:1.46
```

if any volume does not exist, then all snapshot can't be tagged. I suppose it should just ignore missed volume, but tag other snapshot. 

I found we use `paginate` to get volume information, then just pass all volume list of snapshot to `p.paginate` function, then if volume does not exist, this function will report exception. Is it as expected? I suppose we should still tag those snapshot that can find volume information.

**Comment by kapilt** [comment]
can you confirm that you have a snapshot from a present volume, that has a project tag? afaics the behavior with skip_missing: true should bee as you described. ie. whats the total population of snapshots your looking at, also please include the entire log, not just an excerpt with --verbose on the cli (and omit any account ids).

**Comment by zbw1983** [comment]
Thanks for your responds. Yes, I have snapshot from a present volume with project tag. I re-run this program for 2 snapshots, 1 has EBS volume, another doesn't have, below is the whole output, if I only run one snapshot with volume, snapshot can be tagged. 

```2020-02-26 11:26:47,065: custodian.commands:DEBUG Loaded file copy-tag.yml. Contains 1 policies
2020-02-26 11:26:48,927: custodian.output:DEBUG Storing output with <LogFile file://snapshot-tag/copy-tags-from-ebs-volume-to-snapshot/custodian-run.log>
2020-02-26 11:26:48,940: custodian.policy:DEBUG Running policy:copy-tags-from-ebs-volume-to-snapshot resource:ebs-snapshot region:us-west-2 c7n:0.8.46.1
2020-02-26 11:26:48,946: custodian.cache:DEBUG Using cache file /Users/zzzz/.cache/cloud-custodian.cache
2020-02-26 11:26:48,946: custodian.resources.snapshot:DEBUG Using cached c7n.resources.ebs.Snapshot: 2152
2020-02-26 11:26:48,950: custodian.resources.snapshot:DEBUG Filtered from 2152 to 2 snapshot
2020-02-26 11:26:48,950: custodian.policy:INFO policy:copy-tags-from-ebs-volume-to-snapshot resource:ebs-snapshot region:us-west-2 count:2 time:0.01
2020-02-26 11:26:48,958: custodian.cache:DEBUG Using cache file /Users/zzzz/.cache/cloud-custodian.cache
2020-02-26 11:26:50,251: custodian.resources.ebs:WARNING event ids not resolved: ['vol-ffffffff', 'vol-03c9620510ab2efb9'] error:An error occurred (InvalidVolume.NotFound) when calling the DescribeVolumes operation: The volume 'vol-ffffffff' does not exist.
2020-02-26 11:26:50,298: custodian.actions:INFO Tagged 0 resources from related, missing-skipped 2 unchanged 0
2020-02-26 11:26:50,298: custodian.policy:INFO policy:copy-tags-from-ebs-volume-to-snapshot action:copyrelatedresourcetag resources:2 execution_time:1.35
2020-02-26 11:26:50,299: custodian.output:DEBUG metric:ResourceCount Count:2 policy:copy-tags-from-ebs-volume-to-snapshot restype:ebs-snapshot scope:policy
2020-02-26 11:26:50,299: custodian.output:DEBUG metric:ApiCalls Count:1 policy:copy-tags-from-ebs-volume-to-snapshot restype:ebs-snapshot```

Seems the codes just send all volume ID to AWS "client.get_paginator.paginate" function, no matter whether it exist or not, then it return error if any volume does not exist.

**Comment by nitrocode** [comment]
I just saw this error with the `mark-for-op` action. I'd also prefer it if the volume doesn't exist, skip it if it doesn't exist.

> [ERROR]	2020-03-13T06:32:41.708Z	8d59c58d-0f62-4680-snip-snap	Exception with tags: [{'Key': 'maid_status', 'Value': 'Resource does not meet policy: delete@2020/04/12'}]  An error occurred (InvalidVolume.NotFound) when calling the CreateTags operation: The volume 'vol-snip' does not exist.

**Comment by zbw1983** [comment]
yes, it should skip the not exist volume, but it should not stop the whole process, or the user can't add tag to hundreds of snapshots if any volume is missed.

**Comment by dvianello** [comment]
Having the same issue here, which basically means we are unable to tag snapshots. 

Confirmed with a brand new snapshot from a new EBS volume of a new instance.

**Comment by nitrocode** [comment]
The easiest fix may be to wrap a `try` around each `create_tag` call

https://github.com/cloud-custodian/cloud-custodian/blob/00906a061341688cf7d19245d3a0126422ccfa82/c7n/resources/ebs.py#L536-L537

https://github.com/cloud-custodian/cloud-custodian/blob/00906a061341688cf7d19245d3a0126422ccfa82/c7n/resources/ebs.py#L1055-L1061

https://github.com/cloud-custodian/cloud-custodian/blob/00906a061341688cf7d19245d3a0126422ccfa82/c7n/resources/ebs.py#L1280-L1283

https://github.com/cloud-custodian/cloud-custodian/blob/00906a061341688cf7d19245d3a0126422ccfa82/c7n/resources/ebs.py#L1294-L1297

https://github.com/cloud-custodian/cloud-custodian/blob/00906a061341688cf7d19245d3a0126422ccfa82/c7n/resources/ebs.py#L1308-L1311

**Comment by nitrocode** [comment]
What's interesting is that there are multiple catches for `InvalidVolume.NotFound` so there must be one area that's missing.
